yt/visualization/base_plot_types.py

```diff
@@ -217,34 +217,43 @@ def __init__(self, fsize, axrect, caxrect, zlim, figure, axes, cax):
 
     def _init_image(self, data, cbnorm, cblinthresh, cmap, extent, aspect):
         """Store output of imshow in image variable"""
+        cbnorm_kwargs = dict(
+            vmin=float(self.zmin) if self.zmin is not None else None,
+            vmax=float(self.zmax) if self.zmax is not None else None,
+        )
         if cbnorm == "log10":
-            norm = matplotlib.colors.LogNorm()
+            cbnorm_cls = matplotlib.colors.LogNorm
         elif cbnorm == "linear":
-            norm = matplotlib.colors.Normalize()
+            cbnorm_cls = matplotlib.colors.Normalize
         elif cbnorm == "symlog":
             if cblinthresh is None:
                 cblinthresh = float((np.nanmax(data) - np.nanmin(data)) / 10.0)
 
-            symlog_kwargs = dict(
-                linthresh=cblinthresh,
-                vmin=float(np.nanmin(data)),
-                vmax=float(np.nanmax(data)),
+            cbnorm_kwargs.update(
+                dict(
+                    linthresh=cblinthresh,
+                    vmin=float(np.nanmin(data)),
+                    vmax=float(np.nanmax(data)),
+                )
```

Comment:
In my dream where we are already all using Python 3.9
```suggestion
            cbnorm_kwargs |= dict(
                linthresh=cblinthresh,
                vmin=float(np.nanmin(data)),
                vmax=float(np.nanmax(data)),
            )
```

Reply:
Yeah I actually wrote this _while listening to a podcast on Python 3.9.0_... I had to resist the urge of doing exactly this :D